pkg/elasticetcd/doc.go

```diff
@@ -0,0 +1,24 @@
+// Package elasticetcd implements an autoscaling etcd cluster.
+// This package was inspired by, and implements the elastic scaling algorithm as described in, the github.com/purpleidea/mgmt/etcd package.
+//
+// The elastic scaling algorithm is as follows,
+// 	- The elasticetcd instance is started, and can be passed a list of existing etcd endpoints
+//  	- If no endpoints are given, assume you are the first instance up and start your embedded etcd server
+//	- Connect to the given endpoints (or your own etcd endpoints if you were the first one up)
+//	- Volunteer to be a server, and wait to be nominated as a server
```

Comment:
Yes, all elasticetcd instances will volunteer to be a server. As a package, I would like elasticetcd to continue this way, and leave out the choice to be/not be a server to whoever uses the package. Like how the store package does in this PR. Without the server part, elasticetcd is just a etcd client, so you're better off just using etcd client directly. If elasticetcd were to become a standalone app, then that app could provide such a feature.

But, as mentioned in one of the TODOs below, I would also like to have the ability to use different server selection algorithms available for elasticetcd.